app/src/main/java/org/schabi/newpipe/player/gesture/MainPlayerGestureListener.kt

```diff
@@ -0,0 +1,224 @@
+        }
+
+        binding.volumeImageView.setImageDrawable(
+            AppCompatResources.getDrawable(
+                player.context,
+                when {
+                    currentProgressPercent <= 0 -> R.drawable.ic_volume_off
+                    currentProgressPercent < 0.25 -> R.drawable.ic_volume_mute
+                    currentProgressPercent < 0.75 -> R.drawable.ic_volume_down
+                    else -> R.drawable.ic_volume_up
+                }
+            )
+        )
+
+        if (binding.volumeRelativeLayout.visibility != View.VISIBLE) {
+            binding.volumeRelativeLayout.animate(true, 200, AnimationType.SCALE_AND_ALPHA)
+        }
+        if (binding.brightnessRelativeLayout.visibility == View.VISIBLE) {
+            binding.volumeRelativeLayout.visibility = View.GONE
+        }
+    }
+
+    private fun onScrollBrightness(distanceY: Float) {
+        val parent: AppCompatActivity = playerUi.parentActivity.orElse(null) ?: return
+        val window = parent.window
+        val layoutParams = window.attributes
+        val bar: ProgressBar = binding.brightnessProgressBar
+        val oldBrightness = layoutParams.screenBrightness
+        bar.progress = (bar.max * max(0f, min(1f, oldBrightness))).toInt()
+        bar.incrementProgressBy(distanceY.toInt())
+        val currentProgressPercent = bar.progress.toFloat() / bar.max
+        layoutParams.screenBrightness = currentProgressPercent
+        window.attributes = layoutParams
+
+        // Save current brightness level
+        PlayerHelper.setScreenBrightness(parent, currentProgressPercent)
+        if (DEBUG) {
+            Log.d(
+                TAG,
+                "onScroll().brightnessControl, " +
+                    "currentBrightness = " + currentProgressPercent
+            )
+        }
+        binding.brightnessImageView.setImageDrawable(
+            AppCompatResources.getDrawable(
+                player.context,
+                if (currentProgressPercent < 0.25) R.drawable.ic_brightness_low else if (currentProgressPercent < 0.75) R.drawable.ic_brightness_medium else R.drawable.ic_brightness_high
+            )
+        )
+        if (binding.brightnessRelativeLayout.visibility != View.VISIBLE) {
+            binding.brightnessRelativeLayout.animate(true, 200, AnimationType.SCALE_AND_ALPHA)
+        }
+        if (binding.volumeRelativeLayout.visibility == View.VISIBLE) {
+            binding.volumeRelativeLayout.visibility = View.GONE
+        }
+    }
+
+    override fun onScrollEnd(event: MotionEvent) {
+        super.onScrollEnd(event)
+        if (binding.volumeRelativeLayout.visibility == View.VISIBLE) {
+            binding.volumeRelativeLayout.animate(false, 200, AnimationType.SCALE_AND_ALPHA, 200)
+        }
+        if (binding.brightnessRelativeLayout.visibility == View.VISIBLE) {
+            binding.brightnessRelativeLayout.animate(false, 200, AnimationType.SCALE_AND_ALPHA, 200)
+        }
+    }
```

Comment:
Could ``visibility == View.VISIBLE`` be replaced by ``isVisible``? (applies to whole class)